x/twap/api_test.go

```diff
@@ -837,3 +841,70 @@ func (s *TestSuite) TestGetArithmeticTwapToNow_ThreeAsset() {
 		})
 	}
 }
+
+func (s *TestSuite) TestGeometricTwapToNow_BalancerPool_Randomized() {
+	seed := int64(1)
+	r := rand.New(rand.NewSource(seed))
+	retries := 1000
+
+	type randTestCase struct {
+		elapsedTimeMs int
+		weightA       sdk.Int
+		tokenASupply  sdk.Int
+		weightB       sdk.Int
+		tokenBSupply  sdk.Int
+	}
+
+	maxUint64 := ^uint(0)
+
+	for i := 0; i < retries; i++ {
+		elapsedTimeMs := sdkrand.RandIntBetween(r, 1, int(maxUint64>>1))
+		weightA := sdk.NewInt(int64(sdkrand.RandIntBetween(r, 1, 1000)))
+		tokenASupply := sdk.NewInt(int64(sdkrand.RandIntBetween(r, 10_000, 1_000_000_000)))
+
+		tokenBSupply := sdk.NewInt(int64(sdkrand.RandIntBetween(r, 10_000, 1_000_000_000)))
+		weightB := sdk.NewInt(int64(sdkrand.RandIntBetween(r, 1, 1000)))
+
+		s.Run(fmt.Sprintf("elapsedTimeMs=%d, weightA=%d, tokenASupply=%d, weightB=%d, tokenBSupply=%d", elapsedTimeMs, weightA, tokenASupply, weightB, tokenBSupply), func() {
+			s.SetupTest()
+
+			ctx := s.Ctx
+			app := s.App
+
+			assets := []balancer.PoolAsset{
+				{
+					Token:  sdk.NewCoin(denom0, tokenASupply),
+					Weight: weightA,
+				},
+				{
+					Token:  sdk.NewCoin(denom1, tokenBSupply),
+					Weight: weightB,
+				},
+			}
+
+			s.PrepareCustomBalancerPool(assets, balancer.PoolParams{
+				SwapFee: sdk.ZeroDec(),
+				ExitFee: sdk.ZeroDec(),
+			})
+
+			//
```

Comment:
Were you going to add a comment here?

Reply:
Updated:

```
// We add 1ms to avoid always landing on the same block time
// In that case, the most recent spot price would be used
// instead of interpolation.
```

I think that answers your other question as well